### PR TITLE
Use Solidity's new style constructors

### PR DESCRIPTION
**Pull request by raininja**
I've implemented the rule for new style constructor, submitting a PR as requested by @duaraghav8

### CODE REVIEW

**Comment by duaraghav8** [comment]
@raininja Thanks for submitting the PR. I'm hoping Scott will assign the gitcoin bounty to you soon since you worked for it and deserve it.
Reviewing it.

File: docs/developer-guide.rst [hunk 1]
```diff
@@ -587,10 +591,10 @@ Inside your main plugin directory itself:
 			returnInternalIssues: true
 		}
 	};
-	describe ('Rule foo: Acceptances', function () {
-		it ('should accept some stuff and reject other stuff', function (done) {
-			var code = 'contract BlueBerry { function foo () {} }';
-			var errors = Solium.lint (code, config);
+	describe ('Rule foo: Acceptances', () => {
+		it ('should accept some stuff and reject other stuff', done => {
+			const code = 'contract BlueBerry { function foo () {} }';
```
Comment (eayoungs) [inline docs/developer-guide.rst]: I know this is unsolicited but I'm eager to learn and found this PR through Gitcoin.  So, assuming you're willing to humor me; shouldn't `function foo ()` be changed to `constructor()`?

**Comment by duaraghav8** [comment]
@raininja can you please re-fork the latest develop and add your rule files to it and then send in a PR? The last PR i merged also included a lot of branch conflict fixing that I did. It would make the review process easier for me :)

**Comment by raininja** [comment]
@duaraghav8  whoops, sorry about that, I see that I've left some cruft in this PR.  Will do right away closing this one.

**Comment by raininja** [comment]
@duaraghav8  there seems to be an issue with running npm install on the latest clone. . .  please let me know how to proceed.  Here is the error: 

`[raijin@ishtar][~/workspace/Solium]%npm install --dev
npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/chai-fd85cc3e/index.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/chai-fd85cc3e/karma.conf.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/chai-fd85cc3e/karma.sauce.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/chai-fd85cc3e/sauce.browsers.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/acorn-c0503b61/dist/acorn.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/chai-fd85cc3e/ReleaseNotes.md'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/eslint-15702fc8/LICENSE'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/eslint-15702fc8/README.md'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/ajv-532715fc/dist/ajv.min.js'
npm WARN tar ENOENT: no such file or directory, open '/home/raijin/workspace/Solium/node_modules/.staging/mocha-fcdb810d/CHANGELOG.md'
npm ERR! path /home/raijin/.npm/_cacache/content-v2/sha512/9d/8d/d6e46bb6ada72f74393ffc42d1798f9d1e36fd3e521c5431573fbda754886ad86cb72ee5576302683d5b9dc109efd6c86da21a6091d08a4f661a298d022b
npm ERR! code EINVAL
npm ERR! errno -22
npm ERR! syscall lstat
npm ERR! EINVAL: invalid argument, lstat '/home/raijin/.npm/_cacache/content-v2/sha512/9d/8d/d6e46bb6ada72f74393ffc42d1798f9d1e36fd3e521c5431573fbda754886ad86cb72ee5576302683d5b9dc109efd6c86da21a6091d08a4f661a298d022b'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/raijin/.npm/_logs/2018-06-28T04_30_55_467Z-debug.log
`

**Comment by raininja** [comment]
@duaraghav8  I've tried to clean my npm cache and it's [worse](https://hastebin.com/isesapizem.sql)

**Comment by duaraghav8** [comment]
@raininja I'm not sure how you're setting up solium in your workstation, below is my log:

```
$ git clone https://github.com/duaraghav8/Solium.git
Cloning into 'Solium'...
remote: Counting objects: 4967, done.
remote: Compressing objects: 100% (28/28), done.
remote: Total 4967 (delta 14), reused 26 (delta 8), pack-reused 4931
Receiving objects: 100% (4967/4967), 1.18 MiB | 950.00 KiB/s, done.
Resolving deltas: 100% (3247/3247), done.
raghavdua at macbook-pro in ~/Desktop/scratchpad/node/test
$
raghavdua at macbook-pro in ~/Desktop/scratchpad/node/test
$ cd Solium/
raghavdua at macbook-pro in ~/Desktop/scratchpad/node/test/Solium on master*
$ gs
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
raghavdua at macbook-pro in ~/Desktop/scratchpad/node/test/Solium on master*
$ git checkout develop
Branch 'develop' set up to track remote branch 'develop' from 'origin'.
Switched to a new branch 'develop'
raghavdua at macbook-pro in ~/Desktop/scratchpad/node/test/Solium on develop*
$ npm i --dev
npm WARN install Usage of the `--dev` option is deprecated. Use `--only=dev` instead.

> fsevents@1.1.2 install /Users/raghavdua/Desktop/scratchpad/node/test/Solium/node_modules/fsevents
> node install

[fsevents] Success: "/Users/raghavdua/Desktop/scratchpad/node/test/Solium/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile
added 609 packages in 6.009s
```

clone your new fork, checkout into develop, then `npm i --dev`. This should do it

**Comment by raininja** [comment]
looks like I broke my npm @duaraghav8  -_-  had I checked my paste more closely, I would have seen that, sorry for thee inconvenience

**Comment by raininja** [comment]
sorry about that @duaraghav8 , unexpected (seemingly filesystem related, f2fs) issue, I've got npm working again

**Comment by raininja** [comment]
wait, I must have misunderstood, which branch did you want me to fork, @duaraghav8 ?  there isn't a "develop" branch

**Pull request closed without merging**